x-pack/plugins/log_explorer/public/utils/resource.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LogDocument, ResourceFields } from '../../common/document';
+import * as constants from '../../common/constants';
+
+type Field = keyof LogDocument['flattened'];
+
+const getFieldFromDoc = <T extends Field>(doc: LogDocument, field: T) => {
+  const fieldValueArray = doc.flattened[field];
+  return fieldValueArray && fieldValueArray.length ? fieldValueArray[0] : undefined;
+};
+
+export const getUnformattedResourceFields = (doc: LogDocument): ResourceFields => {
+  const serviceName = getFieldFromDoc(doc, constants.SERVICE_NAME_FIELD);
+  const hostName = getFieldFromDoc(doc, constants.HOST_NAME_FIELD);
+  const agentName = getFieldFromDoc(doc, constants.AGENT_NAME_FIELD);
+  const orchestratorClusterName = getFieldFromDoc(doc, constants.ORCHESTRATOR_CLUSTER_NAME_FIELD);
+  const orchestratorResourceId = getFieldFromDoc(doc, constants.ORCHESTRATOR_RESOURCE_ID_FIELD);
+  const orchestratorNamespace = getFieldFromDoc(doc, constants.ORCHESTRATOR_NAMESPACE_FIELD);
+  const containerName = getFieldFromDoc(doc, constants.CONTAINER_NAME_FIELD);
+  const containerId = getFieldFromDoc(doc, constants.CONTAINER_ID_FIELD);
+  const cloudInstanceId = getFieldFromDoc(doc, constants.CLOUD_INSTANCE_ID_FIELD);
+
+  return {
+    [constants.SERVICE_NAME_FIELD]: serviceName,
```

Comment:
Is the order here defining what the priority of each field over the other is? Or where is the order defined?

Reply:
The order of rendering is defined at the presentation layer here - https://github.com/elastic/kibana/blob/6e7f9a0c9aab43e0255676ba06ab0b7ec5699e2c/x-pack/plugins/log_explorer/public/components/virtual_columns/resource.tsx#L24

We simply check for field present or not and then render it.

Reply:
Ok, good enough for now. When we start to tweaking this, it would be nice to just have a list of the fields and moving around the field in the list changes automatically in which order it is rendered (for later).

Reply:
Sounds good. I initially thought of doing so but then anyhow this logic will go away and we have a API which will return us the Resources. We can wait for it and then see how that would look like